### PR TITLE
Use log value of strain rate and stress in computing exact Peierls vi…

### PR DESCRIPTION
**Pull request by lhy11009**
Pull Request Checklist. Please read and check each box with an X. Delete any part not applicable. Ask on the [forum](https://community.geodynamics.org/c/aspect) if you need help with any step.

In this pull request, I modified a previous "compute_exact_strain_rate_and_derivative". This is used for computing the value of the Peierls strain rate and the derivative to stress. These values are then used in the iteration of the "exact" scheme of the Peierls viscosity. The modification uses the logarithmic values of the strain rate and the stress, leading to regular ranges of the iteration variables and faster convergence. 
The improvement could be illustrated by comparing a change of the strain rate by 7 orders of magnitude from $10^{-22}$ to $10^{-15}$ with the logarithmic value only varying from -22 to -15. The improvement in convergence is demonstrated by smaller total iteration numbers (in the exact Peierls rheology) from computations of pre-existing tests.

Another advantage of the new method is excluding the prefactor from the computing of the derivatives as the prefactor is separated once the logarithmic values are taken on both sides of the flow law (strain rate to stress). As the prefactor could vary by order of magnitudes in different flow laws, we could avoid potential issues in float limit with the new implementation.
In addition, the residual defined in the new implementation would be the relative residual of the strain rate. Previously, the residual is an absolute residual, which is less consistent with varying strain rates in the model.

Tests:

- fix two previous tests:

1. visco_plastic_peierls_cutoff_test
2. visco_plastic_peierls_mei_exact

During the fix, I make sure the same value of viscosity is calculated as previously.

- add another test:

1. visco_plastic_peierls_idrissi_exact

In this test, I add another case for the Idrissi 2016 flow law

### Before your first pull request:

* [x] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.

### For all pull requests:

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have [created a testcase](https://aspect-documentation.readthedocs.io/en/latest/user/extending/testing/writing-tests.html) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

**Comment by lhy11009** [comment]
@bobmyhill Could you provide a review, thanks.

File: doc/modules/changes/20230708_haoyuan [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Improved: The iteration scheme in the Peierls creep rheology now uses the derivative of the logarithmic of the strain rate to the logarithmic of the stress.
```
Comment (naliboff) [inline doc/modules/changes/20230708_haoyuan:1]:
```suggestion
Improved: The iteration scheme in the Peierls creep rheology now uses the derivative of the logarithm of the strain rate to the logarithm of the stress.
```

**Review by naliboff — COMMENTED** [review]
@lhy11009 - nice work! Tomorrow I will do a more careful read through the modifications to peirls_creep.cc, but overall the PR is in great shape.

Since it is now output in the log file, do you  have a sense for how much of a speedup using the log of strain rate and stress produces for the local iterations? In either case, hopefully this leads to a significant speedup for the other local iterative schemes!

File: tests/visco_plastic_peierls_cutoff_test/screen-output [hunk 2]
```diff
@@ -1,11 +1,15 @@
+-----------------------------------------------------------------------------
```
Comment (bobmyhill) [inline tests/visco_plastic_peierls_cutoff_test/screen-output:1]: These lines can be removed from the test outputs

**Review by bobmyhill — CHANGES_REQUESTED** [review]
Very nice, thank you for doing this! 

Just a couple of suggestions and comments, then this is ready to merge.

File: tests/visco_plastic_peierls_mei_exact.prm [hunk 3]
```diff
@@ -95,7 +95,7 @@ subsection Material model
     set Peierls fitting parameters                = 0.15
     set Peierls glide parameters p                = 0.5
     set Peierls glide parameters q                = 1.0
-    set Peierls strain rate residual tolerance = 1e-22
+    set Peierls strain rate residual tolerance = 1e-6
```
[the remark this answers is not in the document]
Comment (lhy11009) [reply tests/visco_plastic_peierls_mei_exact.prm]: No, there is no reason to it. I have changed the tolerance to 1e-12

File: tests/visco_plastic_peierls_cutoff_test/screen-output (hunk 2 above)
Comment (lhy11009) [reply tests/visco_plastic_peierls_cutoff_test/screen-output]: Yeah, you are right, I have removed them.

**Review by lhy11009 — COMMENTED** [review]
1. Commit the suggested changes
2. Fix some values of tolerance used in tests & fix the redundant outputs in the tests.

Answer to John's question:
https://github.com/geodynamics/aspect/pull/5227#pullrequestreview-1520834291

1. For the Peierls rheology we are applying this right now, this only decreases the number of iterations by 1-2 in the case of the Idrissi flow law. And for the Mei 10 flow law, the number of iterations would remain the same. This has to do with the approximate Peierls viscosity being very good and the strain rate varies little in the iteration.
2. In the Case of the Idrissi flow law, we have observed improvement in the first few iterations where the strain rate needs to vary by magnitudes. In this case, the new scheme is much more efficient that the residual drops by magnitude immediately. This is expected, since by taking the log values we are getting the magnitudes so that we are dealing with them directly.
3. Yes, the new scheme has the potential to speed up other iterations of non-linear rheology. We should discuss where to test it after this PR.

**Review by bobmyhill — APPROVED** [review]

**Comment by bobmyhill** [comment]
Because the original strain rate and derivative functions are used in other parts of the code, you'll have to re-add them in addition to your new functions - we'll remove them in a later PR.

**Comment by lhy11009** [comment]
Update:
1. Add back the original function
2. Checked build and Peierls creep tests

**Pull request merged**